dev-docs/RFCs/v4.0.0/terms_for_concepts_and_datatypes.md

```diff
@@ -0,0 +1,96 @@
+    - an H3 cell which is a **topological** pentagon
+- `DirectedEdge`
+    - an H3 index of `mode 2`
+    - representing a traversal from an origin cell to an adjacent destination cell
+
+### Notes
+
+We realize that "hexagon" will still be used informally to refer to the concept of "cell" (As the development team, we do it ourselves!). This should be expected in casual, informal discussions of H3. When **precision** is required, however, we advise the use of strict technical terms like "index", "cell", "hexagon", "pentagon", etc.
+
+
+## Use of "Topological"
+
+We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## Names for H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
+
+
+## Function Names
+
+A few more function name change proposals:
+
+|          Current name         |     Proposed name     |                  Notes                  |
+|-------------------------------|-----------------------|-----------------------------------------|
+| (doesn't exist)               | `isValidIndex`        |                                         |
+| `h3IsValid`                   | `isValidCell`         |                                         |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |                                         |
+| `h3IsPentagon`                | `isPentagon`          |                                         |
+| `h3ToParent`                  | `cellToParent`        |                                         |
+| `h3ToChildren`                | `cellToChildren`      |                                         |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |                                         |
+| `geoToH3`                     | `geoToCell`           |                                         |
+| `h3ToGeo`                     | `cellToGeo`           |                                         |
+| `kRing`                       | `disk`                | filled-in disk                          |
```

Comment:
Using `neighbors` for `disk` would conflict with the use of `neighbors` in `cellsAreNeighbors`.

I think `cellsWithinRadius` and `cellsAtRadius` would be more descriptive than the names we have now, but I do like the concision of `ring` and `disk`.

Also, thinking about how many verbal conversations I already have around "k-rings", it would be less of a mouthful to say `ring` and `disk` than the `cells*Radius` names. Thinking about how many times I've said, "Do you mean a hollow ring, or a filled-in ring?", it'd be great to have short, precise names for these concepts.